### ISSUE TITLE
Test new version of Declearn (2.8) with Fed-BioMed

### ISSUE THREAD

**Issue by srcansiz** [fedbiomed/fedbiomed#1667]
New version of DecLearn is released. There are slight changes in the Optimizers. According to DecLearn team these changes does not break the API. I propose to create a new branch and test DecLearn 2.8 compatibility. 

- [x] Test running advance optimizer user guide and tutorials 
- [x] Use e2e tests validate `declearn` 2.8 version and add new e2e tests if possible.

**Comment by alitolga** [comment]
I encountered a bug on "Declearn-with-Sklearn" tutorial notebook. Mainly:

```
2026-03-19 17:40:02,048 fedbiomed ERROR [_sklearn_models._parse_sample_losses:230] - Value error during monitoring: could not convert string to float: '0.000000, Objective: 0.000000'
```

I found out that we rely on the stdout output of `self.model.partial_fit()` in `sklearn.py::BaseSkLearnModel::train() lines 266-267` which gets transmitted to sklearn_models.py::SKLearnTrainingPlanPartialFit::_parse_sample_losses() incorrectly. `Lines 224-230:`

```python
split = row.rsplit("loss: ", 1)
if len(split) == 1:  # no "loss: " in the line
    continue
try:
    losses.append(float(split[1]))
except ValueError as exc:
    logger.error(f"Value error during monitoring: {exc}")
```

guard against an stdout of type "Loss: 0.123", but not for other outputs.

I am not sure if this is directly caused by the declearn optimizer, or directly from the sklearn models. For now, I am planning to patch as:

``` python
loss_str = split[1].split(",", 1)[0].strip()
try:
    losses.append(float(split[1]))
```

and continue testing the notebooks. However, I would be leaned towards changing the approach in the future, as directly depending on the stdout feels less reliable.

**Comment by srcansiz** [comment]
Hi @alitolga, 

Thank you for this finding. I remember that we faced similar issue before. Just to make sure,  can you please try
1.  Without declearn optimizers
2. Without upgrading but using  declearn optimizer. 

This can also gives us a clue if new upgraded declearn optimizer return loss value in different type.

**Comment by srcansiz** [comment]
> I encountered a bug on "Declearn-with-Sklearn" tutorial notebook. Mainly:
> 
> ```
> 2026-03-19 17:40:02,048 fedbiomed ERROR [_sklearn_models._parse_sample_losses:230] - Value error during monitoring: could not convert string to float: '0.000000, Objective: 0.000000'
> ```
> 
> I found out that we rely on the stdout output of `self.model.partial_fit()` in `sklearn.py::BaseSkLearnModel::train() lines 266-267` which gets transmitted to sklearn_models.py::SKLearnTrainingPlanPartialFit::_parse_sample_losses() incorrectly. `Lines 224-230:`
> 
> split = row.rsplit("loss: ", 1)
> if len(split) == 1:  # no "loss: " in the line
>     continue
> try:
>     losses.append(float(split[1]))
> except ValueError as exc:
>     logger.error(f"Value error during monitoring: {exc}")
> guard against an stdout of type "Loss: 0.123", but not for other outputs.
> 
> I am not sure if this is directly caused by the declearn optimizer, or directly from the sklearn models. For now, I am planning to patch as:
> 
> loss_str = split[1].split(",", 1)[0].strip()
> try:
>     losses.append(float(split[1]))
> and continue testing the notebooks. However, I would be leaned towards changing the approach in the future, as directly depending on the stdout feels less reliable.

and yes if there is better approach to retrieve loss values please do not hesitate create a new issue for that.

**Comment by srcansiz** [comment]
Hi @alitolga ,

Could you please share your updates  concerning these two tests:
> Without declearn optimizers
> Without upgrading but using declearn optimizer.

**Comment by alitolga** [comment]
> * Without declearn optimizers
> * Without upgrading but using  declearn optimizer.

Hi @srcansiz ,

I can confirm the bug exists for these cases as well. I can also confirm the bug is fixed with both cases with the patch.

I can continue with the tests.
